Error while deploying using Docker (both methods)

Hi! We had a short exchange about this on another issue (#51). Don't know if you are working on this but I tested today and the installation with Docker (with data in container) still gives this error:

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 7.643 s
[INFO] Finished at: 2018-10-15T08:36:26+00:00
[INFO] Final Memory: 17M/165M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project matchbox: Could not resolve dependencies for project org.broadinstitute.macarthurlab:matchbox:jar:0.1.0: The following artifacts could not be resolved: org.monarchinitiative.exomiser:exomiser-spring-boot-starter:jar:10.1.0, org.monarchinitiative.exomiser:exomiser-spring-boot-test:jar:10.1.0: Could not find artifact org.monarchinitiative.exomiser:exomiser-spring-boot-starter:jar:10.1.0 in central (https://repo.maven.apache.org/maven2) -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
The command '/bin/sh -c $MVN -Dmaven.test.skip=true clean install package' returned a non-zero code: 1

I'll test with other Docker method now!

@northwestwitch 

Hello.

I also got same error message during updating new release matchbox.

The used matchbox version is **with_data_mounted_to_container** and the reference directory details are as follows:
ls -al /data/reference_data/
total 16
drwxr-xr-x 4 rarebird rarebird 4096 11월  8 18:08 .
drwxr-xr-x 3 root     root     4096 11월  8 18:13 ..
drwxrwx--- 3 rarebird rarebird 4096  7월 27 12:40 1807_phenotype
drwxrwxrwx 3 rarebird rarebird 4096  8월  7  2017 exomiser-cli-8.0.0

When docker image is building, I got the same error for "cannot find symbol , symbol:   variable Mockito" as follows. I also got same error when I tried to install matchbox using another docker version image.


[INFO] 4 errors
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 59.763 s
[INFO] Finished at: 2018-11-08T09:28:36+00:00
[INFO] Final Memory: 40M/329M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project matchbox: Compilation failure: Compilation failure:
[ERROR] /Exomiser/matchbox/src/main/java/org/broadinstitute/macarthurlab/matchbox/config/ExomiserConfig.java:[3,19] package org.mockito does not exist
[ERROR] /Exomiser/matchbox/src/main/java/org/broadinstitute/macarthurlab/matchbox/config/ExomiserConfig.java:[23,17] cannot find symbol
[ERROR] symbol:   variable Mockito
[ERROR] location: class org.broadinstitute.macarthurlab.matchbox.config.ExomiserConfig
[ERROR] /Exomiser/matchbox/src/main/java/org/broadinstitute/macarthurlab/matchbox/config/ExomiserConfig.java:[24,17] cannot find symbol
[ERROR] symbol:   variable Mockito
[ERROR] location: class org.broadinstitute.macarthurlab.matchbox.config.ExomiserConfig
[ERROR] /Exomiser/matchbox/src/main/java/org/broadinstitute/macarthurlab/matchbox/config/ExomiserConfig.java:[25,17] cannot find symbol
[ERROR] symbol:   variable Mockito
[ERROR] location: class org.broadinstitute.macarthurlab.matchbox.config.ExomiserConfig
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
The command '/bin/sh -c $MVN -Dmaven.test.skip=true clean install package' returned a non-zero code: 1

--- a/deploy/docker/with_data_in_container/Dockerfile
+++ b/deploy/docker/with_data_in_container/Dockerfile
@@ -9,7 +9,7 @@ RUN apt-get update \
  && apt-get install -y python-dev \
  && apt-get install -y python-setuptools \
  && /usr/bin/easy_install -U pip \
- && /usr/local/bin/pip install crcmod 
+ && /usr/local/bin/pip install crcmod
 
 MAINTAINER MacArthur Lab
 
@@ -52,7 +52,7 @@ RUN wget https://storage.googleapis.com/pub/gsutil.tar.gz \
 #----now get the data and untar it
 
 WORKDIR data
-RUN /root/gsutils_dir/gsutil/gsutil -m -o GSUtil:parallel_composite_upload_threshold=150M cp gs://seqr-reference-data/1711_phenotype.tar.gz data.local.tar.gz \
+RUN /root/gsutils_dir/gsutil/gsutil -m -o GSUtil:parallel_composite_upload_threshold=150M cp gs://seqr-reference-data/1807_phenotype.tar.gz data.local.tar.gz \
  && tar -xzf data.local.tar.gz \
  && rm data.local.tar.gz \
  && pwd \
@@ -71,13 +71,13 @@ RUN cp -rf $MATCHBOX_CONFIG_DIR . \
 #                                           #
 # Please note the EXOMISER_DATA_DIR         #
 # value. The file system path with ref      #
-# data (viewable by docker daemon) must     #  
+# data (viewable by docker daemon) must     #
 # be mounted to this location in            #
 # container at the docker run step          #
 #                                           #
 #############################################
 env EXOMISER_DATA_DIR=/Exomiser/matchbox/data
-env EXOMISER_PHENOTYPE_DATA_VERSION=1711
+env EXOMISER_PHENOTYPE_DATA_VERSION=1807
 
 
 #############################################
@@ -136,7 +136,7 @@ env USE_HTTPS=false
 #					-keystore $HTTPS_SSL_KEY_STORE \
 #					-storepass $HTTPS_SSL_KEY_STORE_PASSWORD \
 #					-keypass $HTTPS_SSL_KEY_PASSWORD
-					
+
 
 #############################################
 #                                           #

--- a/deploy/docker/with_data_mounted_to_container/Dockerfile
+++ b/deploy/docker/with_data_mounted_to_container/Dockerfile
@@ -3,7 +3,7 @@ FROM openjdk:8-jdk
 RUN apt-get update \
  && apt-get install -y wget \
  && apt-get install -y git \
- && apt-get install -y maven 
+ && apt-get install -y maven
 
 MAINTAINER MacArthur Lab
 
@@ -42,13 +42,13 @@ RUN cp -rf $MATCHBOX_CONFIG_DIR . \
 #                                           #
 # Please note the EXOMISER_DATA_DIR         #
 # value. The file system path with ref      #
-# data (viewable by docker daemon) must     #  
+# data (viewable by docker daemon) must     #
 # be mounted to this location in            #
 # container at the docker run step          #
 #                                           #
 #############################################
 env EXOMISER_DATA_DIR=/Exomiser/matchbox/data/data
-env EXOMISER_PHENOTYPE_DATA_VERSION=1711
+env EXOMISER_PHENOTYPE_DATA_VERSION=1807
 
 
 #############################################
@@ -107,7 +107,7 @@ env USE_HTTPS=false
 #					-keystore $HTTPS_SSL_KEY_STORE \
 #					-storepass $HTTPS_SSL_KEY_STORE_PASSWORD \
 #					-keypass $HTTPS_SSL_KEY_PASSWORD
-					
+
 
 #############################################
 #                                           #

--- a/deploy/docker/with_data_mounted_to_container/README.md
+++ b/deploy/docker/with_data_mounted_to_container/README.md
@@ -1,9 +1,9 @@
 # <i>matchbox Docker build process (beta)</i>
 
-We are still in the process of designing a best practices build and deployment process using Docker and Kubernetes, so please consider this a work in progress. 
+We are still in the process of designing a best practices build and deployment process using Docker and Kubernetes, so please consider this a work in progress.
 
 Please be careful to not check in Docker files with private secure
-usernames, passwords, and tokens etc. 
+usernames, passwords, and tokens etc.
 
 Please also remember to change any default passwords built into system before production!
 
@@ -14,12 +14,12 @@ Please also remember to change any default passwords built into system before pr
 2. Reference data for the Exomiser dependency fetched from:
 	```
 	Please note, this is a large file
-	
-	wget https://storage.googleapis.com/seqr-reference-data/1711_phenotype.tar.gz
-	
+
+	wget https://storage.googleapis.com/seqr-reference-data/1807_phenotype.tar.gz
+
 	Once you download this file, please uncompress it, and remember the file path. For example,
 	/data/reference_data/
-	
+
 	Also make sure this directory is accessible to your docker daemon since it needs to be mounted
 	via the -v option at docker "run command"
 	```
@@ -30,9 +30,9 @@ Please also remember to change any default passwords built into system before pr
 ## Build process:
 
 1. First update the Dockerfile empty fields at the bottom,
-	
+
 	For example:
-	
+
 	```
 		env MONGODB_HOSTNAME=192.168.1.4
 		env MONGODB_PORT=27017
@@ -40,16 +40,16 @@ Please also remember to change any default passwords built into system before pr
 		env MONGODB_PASSWORD=pwd
 		env MONGODB_DATABASE=mme_primary
 	```
-	
+
 	If you want to serve as HTTPS, please uncomment following by removing "#" and populate as needed. You can ignore otherwise
-	
+
 	```
 	env USE_HTTPS=true
 	env SERVER_PORT=8443
 	env HTTPS_SSL_KEY_STORE=matchbox_keystore
 	env HTTPS_SSL_KEY_STORE_PASSWORD=changeit
 	env HTTPS_SSL_KEY_PASSWORD=<temp_ks_pwd__change_me!>
-	
+
 	RUN keytool -genkey -noprompt \
 						-alias matchbox \
 						-dname "CN=, OU=, O=, L=, S=, C=" \
@@ -58,53 +58,49 @@ Please also remember to change any default passwords built into system before pr
 						-keypass $HTTPS_SSL_KEY_PASSWORD
 	```
 
-	
+
 2. In the deploy/docker directory there are two files that should be handled extra carefully in production given that they will contain tokens and access information for your instance and other nodes.
 	```
-	config.xml : this XML file is used to configure the token to give access to your matchbox instance. 
+	config.xml : this XML file is used to configure the token to give access to your matchbox instance.
 	nodes,json : this JSON file contains tokens that give your matchbox instance access to other MME nodes
-	``` 
-	
-	Using guidance from the example data inside them, populate as needed. 
-	
+	```
+
+	Using guidance from the example data inside them, populate as needed.
+
 	Please remember to remove default values before production!
-	
+
 	Possibly use a secrets-file management system to keep fully populate files that can inserted in at deployment.
-	
-	
+
+
 3. Then, from the matchbox docker directory, do a build (should take 6-10mins max)
 	```
 		docker build -t matchbox-docimg .
 	```
-	
+
 4. Assuming,
 
-	* You have already downloaded the necessary reference data for Exomiser (for example to /data/reference_data/) 
-	
-	* And reference data directory is accessible to Docker daemon, 
-	
-	* And you have a MongoDB instance running and you have added its credentials and details to the Dockerfile before the build step, 
+	* You have already downloaded the necessary reference data for Exomiser (for example to /data/reference_data/)
+
+	* And reference data directory is accessible to Docker daemon,
+
+	* And you have a MongoDB instance running and you have added its credentials and details to the Dockerfile before the build step,
 
 
 	For example, if you are using the default HTTP settings and didn't change any port numbers:
 	```
-		docker run -ti -p 9020:9020 -v "/data/reference_data/":/Exomiser/matchbox/data/data matchbox-docimg 
-	``` 
-	
+		docker run -ti -p 9020:9020 -v "/data/reference_data/":/Exomiser/matchbox/data/data matchbox-docimg
+	```
+
 	OR
-	
+
 	For example, if you uncommented the HTTPS settings and didn't change any HTTPS port numbers:
 	```
-		docker run -ti -p 8443:8443 -v "/reference_data/exomiser-cli/data":/Exomiser/matchbox/data/data matchbox-docimg 
-	``` 
+		docker run -ti -p 8443:8443 -v "/reference_data/exomiser-cli/data":/Exomiser/matchbox/data/data matchbox-docimg
+	```
 
 
 6. You can test your instance with (make sure to adjust the URL "http://localhost:9020/patient/view" with the port you used),
-	
+
 	```
 		curl -X GET -H "X-Auth-Token: abcd" -H "Accept: application/vnd.ga4gh.matchmaker.v1.0+json" -H "Content-Type: application/x-www-form-urlencoded" http://localhost:9020/patient/view
 	```
-
-
-
- 

--- a/deploy/maven/README.md
+++ b/deploy/maven/README.md
@@ -17,7 +17,7 @@
 
 * Clone the Exomiser package
 	- Make sure you have a settings.xml file in your ~/.m2/ directory with the following entry (to activate a local repsitory for Maven to use)
-	
+
 		```
 			<settings>
     			<localRepository>${user.home}/.m3/repository</localRepository>
@@ -41,40 +41,40 @@
 		```git clone https://github.com/macarthur-lab/matchbox```
 
 
-	- Update the following lines in the src/main/resources/application.properties appropriately. 
-	
+	- Update the following lines in the src/main/resources/application.properties appropriately.
+
 		- If you are NOT planning to proxy matchbox behind a HTTPS service, you would have to start server matchbox as HTTPS per MME requirements.
-	
-			- Uncomment and populate the server.ssl.* attributes to start <i>matchbox</i> as HTTPS. 
-		
+
+			- Uncomment and populate the server.ssl.* attributes to start <i>matchbox</i> as HTTPS.
+
 		- The "exomiser.data-directory=" field is required by Exomiser for phenotype matching. This reference data can be fetched by,
-		
+
 		```
-			wget https://storage.googleapis.com/seqr-reference-data/1711_phenotype.tar.gz
-			
+			wget https://storage.googleapis.com/seqr-reference-data/1807_phenotype.tar.gz
+
 		```
-		
+
 		Then unzip the file.
-		
-		Provide the path of the above untar'ed directory to the "exomiser.data-directory="  and "exomiser.phenotype.data-version" fields 
-		
+
+		Provide the path of the above untar'ed directory to the "exomiser.data-directory="  and "exomiser.phenotype.data-version" fields
+
 		For example, if your reference data was unzipped into,
-		/dev/apps/ref_data/1711_phenotype
-		
-		You would populate the fields as such (note: the "1711_phenotype" in not in the path),
-		
+		/dev/apps/ref_data/1807_phenotype
+
+		You would populate the fields as such (note: the "1807_phenotype" in not in the path),
+
 		```
 			exomiser.data-directory=/dev/apps/ref_data
-			exomiser.phenotype.data-version=1711
+			exomiser.phenotype.data-version=1807
 		```
-		
+
 		Next, populate the MongoDB connection fields as per your MongoDB installation.
-		
-	- Now build source files. 
-		
+
+	- Now build source files.
+
 		```mvn clean install package```
-		
-		
+
+
 	- That should create a directory called "target" with an executable JAR file
 
 
@@ -83,16 +83,14 @@
 		```java -jar target/matchbox-<version>.jar```
 
 
-* NOTE: if you would like to change the default port the server listens on (8080), you can either set/use the environment 
+* NOTE: if you would like to change the default port the server listens on (8080), you can either set/use the environment
 variable ```SERVER_PORT``` or add the argument ```--server.port``` after the ```java -jar``` incantation. For example
- 
+
      ```export SERVER_PORT=9020```
-    
+
   or
-       
+
     ```java -jar matchbox-0.1.0.jar --server.port=9020```
 
-  It is similarly possible to change any of the variables contained in the application.properties in this manner. The 
-  latter is usually a better option as this will be application instance specific rather than as a global system variable. 
-  
-  
+  It is similarly possible to change any of the variables contained in the application.properties in this manner. The
+  latter is usually a better option as this will be application instance specific rather than as a global system variable.

--- a/pom.xml
+++ b/pom.xml
@@ -8,7 +8,7 @@
     <version>0.1.0</version>
     <packaging>jar</packaging>
     <name>matchbox</name>
-	
+
     <parent>
         <groupId>org.springframework.boot</groupId>
         <artifactId>spring-boot-starter-parent</artifactId>
@@ -67,7 +67,7 @@
             <version>${exomiser.version}</version>
             <scope>test</scope>
         </dependency>
-        
+
         <dependency>
     		<groupId>org.apache.commons</groupId>
     		<artifactId>commons-math3</artifactId>
@@ -83,14 +83,12 @@
 			<artifactId>commons-email</artifactId>
 			<version>1.4</version>
 		</dependency>
-		
-		<dependency>
-		    <groupId>org.mockito</groupId>
-		    <artifactId>mockito-all</artifactId>
-		    <version>1.9.5</version>
-		    <scope>test</scope>
-		</dependency>
-
+    <!-- https://mvnrepository.com/artifact/org.mockito/mockito-core -->
+    <dependency>
+       <groupId>org.mockito</groupId>
+       <artifactId>mockito-all</artifactId>
+       <version>2.0.2-beta</version>
+    </dependency>
     </dependencies>
 
     <build>
@@ -117,10 +115,3 @@
         </plugins>
     </build>
 </project>
-
-
-
-
-
-
-  

